src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -85,6 +85,15 @@ protected override void ValidatePropertyType(BindingDiagnosticBag diagnostics)
             VerifyOverridesEqualityContractFromBase(this, diagnostics);
         }
 
+        internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
+
+            var compilation = this.DeclaringCompilation;
+            AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
+            Debug.Assert(WellKnownMembers.IsSynthesizedAttributeOptional(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
+        }
```

Comment:
Seeing this code moved (from accessor to property) and looking at the PR this came from ([https://github.com/dotnet/roslyn/pull/57917](https://github.com/dotnet/roslyn/pull/57917), reviewed by Aleksey and Fred), I think the previous decision was to put the attribute on the accessor. Let's keep it that way. Sorry about that.

Reply:
#57917 was intended to be (mostly) refactoring only (no behavioral changes around the attribute  waa introduced). 

I'm more convinced with the attribute being on the property. 

@AlekseyTs @333fred What do you think?

Reply:
I'm not certain what precedent we have for where we put `CompilerGenerated`. For example, in top-level statements, if we generate a closure type we don't put a `CompilerGenerated` on it because the containing type has `CompilerGenerated`. If the algorithm is "put it on the highest possible point", then putting it on the property instead of the accessors makes sense. If that's not the algorithm, then I'd appreciate knowing the algorithm :).

Reply:
Thinking about this more, if we put the attribute on the property, then what happens if someone derives from the record and overrides the accessor (or adds a setter)?
So I think the attribute should be put on the accessor.

Relevant [spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/records.md#equality-members)

Reply:
@jcouv That makes sense since `CompilerGeneratedAttribute` is inherited.

https://github.com/dotnet/runtime/blob/1d751ba8563fc099c6a75c41687d2f282a05f916/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CompilerGeneratedAttribute.cs#L6

Should I include the attribute on the property **if** the property isn't virtual (ie, the containing record is sealed)? or just keep it simple and leave on the accessor only?

Reply:
Always putting it on the accessor only seems fine/simpler with me.

Reply:
That logic makes sense to me as well.

Reply:
I meant @jcouv's logic.